flocx_market/objects/offer.py

```diff
@@ -66,3 +67,41 @@ def get_all_by_project_id(cls, context):
     def expire(self, context):
         self.status = 'expired'
         self.save(context)
+
+    @classmethod
+    def get_all_by_status(cls,
```

Comment:
I'm not a huge fan of updating `get_all_by_status` rather than simply creating a new class method. But I think that's something that can be updated in a subsequent PR, so I don't think you need to change that now.

Reply:
Leaving this one for now then. Do you think I should write test cases for all of this in objects because right now I am testing even this in matcher?

Reply:
Well, I guess ideally there'd be unit tests in tests/unit/objects/test_offer_object.py (or whatever it's called) that tests permutations of this. But that work would go away once it became a database query, so I'm not sure it's worthwhile. @larsks , thoughts?

Reply:
Cool, so I am leaving for now ... also I rebased the PR.

Reply:
I think the code in `flocx_market/objects/offer.py` should have at least basic tests in `tests/unit/objects/test_object_offer.py`, yes.

Reply:
They are already there, the only thing that isn't there is the test for available offers within a timeslot which have no contracts.